easybuild/tools/build_from_specsfile.py

```diff
@@ -0,0 +1,150 @@
+import yaml
+from easybuild.tools.robot import search_easyconfigs
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.base import fancylogger
+
+
+_log = fancylogger.getLogger('specsfile', fname=False)
+
+
+# general specs applicable to all commands
+class Specsfile(object):
+    def __init__(self):
+        self.easybuild_version = None
+        self.robot = False
```

Comment:
`robot` is just one of the possible options that can be passed, so we need to handle this in a more generic way.

Leave it as is for now, and open an issue to follow up on this ("add general support for defining configuration options in easystack files")